apps/passport-server/src/services/telegramService.ts

```diff
@@ -1036,15 +1038,16 @@ export class TelegramService {
         throw new Error(`User did not submit any valid event ids`);
       }
 
-      const eventsByChat = await fetchEventsPerChat(this.context.dbPool);
-      const telegramChatId = findChatByEventIds(eventsByChat, validEventIds);
-      if (!telegramChatId) {
-        throw new Error(
-          `User attempted to use a ticket for events ${validEventIds.join(
-            ","
-          )}, which have no matching chat`
-        );
+      const eventsByChat = await fetchTelegramEventsByChatId(
```

Comment:
Flow:
1. Get the events associated with the chatId the user submitted.
2. Confirm that a) some events exist for that chatId and b) the validEventIds submitted by the user are a subset of the events linked in the DB.
3. This flow is repeated for anonSend verification as well.